tests/test_area.py

```diff
@@ -92,8 +93,9 @@ def test_markets_are_cycled_according_to_market_count(self):
             assert len(self.area.all_markets) == i
 
     def test_delete_past_markets_instead_of_last(self):
+        constants.D3A_TEST_RUN = False
```

Comment:
(Not directly related to the refactoring) I'm a bit confused by this: what is this constant used for? Shouldn't this tell us that we're running tests?

Reply:
This is a badly-named constant from the past, that used to have a more general behavior. The only reason for the existence of this variable at the moment though, is to control if the old markets are deleted once they move to the past on every market cycle, or if the old market object instances will remain in-memory forever (which is something that some tests - unit and integration - actually require). I think it will be much better to rename this variable to the slightly more verbose DELETE_PAST_MARKETS, what do you think ? If you agree I'll rename this in the context of this PR

Reply:
I agree. In general, I think production code shouldn't know anything about tests. In this case however, I don't see many quick alternatives!

Reply:
On second thought, I would like to also ask @hannesdiedrich to see if he recalls the reason why we named this variable this way. The reason for that is because I recalled that we used to call this variable KEEP_PAST_MARKETS, but we renamed it for some reason down the road, and I really do not want to go full circle without a good reason for this.

Reply:
You are right Spyros, we had this in the past. I couldn't recall why we changed that either, but had I quick look into the code. This constant is not only used to keep the past markets but also to keep all bids and offers in the BaseStrategy buffers in order to assess these in the tests later.  Additionally, have a look here `src/d3a/d3a_core/simulation.py:259` where we also use this constant to export the raw results to JSON files. IMO these usages qualifies the variable naming and not `KEEP_PAST_MARKETS`.

Reply:
But still, I would expect a variable named `D3A_TEST_RUN` to always be `True` during tests (and to be set once and for all before all tests). That's why I think that maybe we could find a better name for it (that doesn't have to do with "tests").
Right now this is used to:
1. keep the past markets in memory
2. keep all bids and offers in memory
3. export the raw results to JSON files

I think that (3) is unrelated to the other two. Maybe we could split this variable into two (or event three) variables, like `KEEP_MARKETS_IN_MEMORY`, `KEEP_BIDS_OFFERS_IN_MEMORY`, `EXPORT_RESULTS_TO_JSON`. It's just an idea of course, but it might make the variable less confusing in my opinion.

Reply:
Thanks Hannes for clarifying this! In this case I would be also hesitant to rename this, and I would not like to split it into many variables since the reason that this variable is used is one (exporting to json has the same root cause, that json exporting depends on past state). Also setting it globally for all tests is not really possible because some tests take advantage of the False value of this variable. How about renaming this variable to RETAIN_PAST_MARKET_STRATEGIES_STATE?

Reply:
I am also not in favour of having 3 different variables and making it more complex than it is. In order to fully consider all parts, the variable name should be `RETAIN_PAST_MARKET_STRATEGIES_STATE_EXPORT_RAW_JSON` which is obviously bad. So I vote for keeping the original name and maybe add a verbose comment in the constants module to explain what this constant does.

Reply:
Export raw json is not really needed as part of the name IMO. The only reason why this variable is used in this feature is because some raw JSON files depend on the results from all markets, because they do not accumulate on every market slot, but they are calculated once at the end of the simulation, and expect all market info (trades, bids, offers) to be there for all past markets for the whole simulation.
Having said that, I agree that probably a verbose comment in the constants.py would be sufficient for our purposes.

Reply:
Not blocking of course, but I would still vote for changing the name, to be honest 😄 `D3A_TEST_RUN` is just saying something different and misleading in my opinion. I would be in favor of changing it to `RETAIN_PAST_MARKET_STRATEGIES_STATE` (or even `RETAIN_PAST_MARKETS_STATES` for brevity, if it makes sense).